### PR TITLE
[V3 General] Fix error in [p]userinfo when game is not set

### LINKED ISSUE #1321: [V3 General] Error in command 'userinfo'.
Please be sure to read through other issues as well to make sure what you are suggesting/reporting has not already
been suggested/reported

### Type:

- [ ] Suggestion
- [x] Bug

### Brief description of the problem
When the command [p]userinfo is done, there is an error if the user has no playing stauts. 
If you have a playing status set, the error does not happen and command works as it should. 

### Expected behavior
using the command [p]userinfo should work if the user has or does not have a playing status set. 

### Actual behavior
error log created:
```
[19/02/2018 14:27] ERROR events on_command_error 162: Exception in command 'userinfo'
Traceback (most recent call last):
  File "/home/unseenmagik/MB3/lib/python3.5/site-packages/discord/ext/commands/core.py", line 62, in wrapped
    ret = yield from coro(*args, **kwargs)
  File "/home/unseenmagik/MB3/lib/python3.5/site-packages/redbot/cogs/general/general.py", line 222, in userinfo
    if user.game.type == 0:  # "Playing" status
AttributeError: 'NoneType' object has no attribute 'type'
[19/02/2018 14:27] ERROR events on_command_error 165: Exception in command 'userinfo'
Traceback (most recent call last):
  File "/home/unseenmagik/MB3/lib/python3.5/site-packages/discord/ext/commands/core.py", line 62, in wrapped
    ret = yield from coro(*args, **kwargs)
  File "/home/unseenmagik/MB3/lib/python3.5/site-packages/redbot/cogs/general/general.py", line 222, in userinfo
    if user.game.type == 0:  # "Playing" status
AttributeError: 'NoneType' object has no attribute 'type'
```

### Steps to reproduce

1. do not set a playing status
2.  run the command [p]userinfo or [p]unserinfo <@user>
3.  Error in command 'userinfo'. Check your console or logs for details.
4.

### PR DESCRIPTION
**Pull request by skmendez**
### Type

- [X] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes

Modifies the game-checking lines so that they don't assume `user.game` is not a NoneType object.

Fixes #1321

### CODE REVIEW

**Review by tekulvw — CHANGES_REQUESTED** [review]
There is a possibility of `game` not being set at all. I'd recommend the following:

```py
game = _("Chilling in {} status").format(user.status)
if user.game is None:
    pass
...
```

**Comment by skmendez** [comment]
Clarify what you mean by `game` not being set at all. For all `Member` objects, they have an attribute `game` (which may or may not be `None`). The only condition where my solution would error is if there was a `user.game.type not in [0, 1, 2, 3]`

File: redbot/cogs/general/general.py
Comment (Tobotimus) [inline redbot/cogs/general/general.py]: Bit of trailing whitespace here on 219 could be removed

File: redbot/cogs/general/general.py
Comment (Tobotimus) [inline redbot/cogs/general/general.py]: I prefer Will's solution which sends the default message if `user.game` is None or something we don't expect.

**Review by Tobotimus — CHANGES_REQUESTED** [review]
Otherwise I think this is fine

File: redbot/cogs/general/general.py
Comment (Tobotimus) [reply redbot/cogs/general/general.py]: Wait a second, don't remove the `else` block. I'd rather have it throw an error (like a `ValueError` or `RuntimeError`) or just display the default "chilling" message as Will's example provides.

**Review by Tobotimus — APPROVED** [review]
Will wait on @tekulvw for the final review

**Review by tekulvw — APPROVED** [review]

**Pull request merged**
